[Frederick Pek] iP

frederickpek opened the pull request:
# Duke
> I had a lot of fun coding Duke! - [Frederick Pek](https://github.com/frederickpek) 

Duke helps you track your tasks at hand, so they're free to do more iP/tP increments!

* ToDo Tasks
* Deadline Tasks
* Event Tasks

Why Duke?

1. Its **FREE**! :moneybag:
1. Its _EASY_! :grinning:

You can keep track of which tasks have been done!
- [x] Buy milk
- [x] Buy bread
- [x] sort array
- [x] finish this GFMD task

Here is a ~~random~~ program that prints "Hello World!" The `main` method is shown below!
```java
public class HelloWorld {
    public static void main(String[] args) {
        System.out.println("Hello World!");
    }
}
```

Commit:
Add Gradle support

frederickpek committed:
.gitignore: add *.class

frederickpek committed:
Implement Level-1

frederickpek committed:
Implement Level-2

frederickpek committed:
Implement Level-3

frederickpek committed:
Implement Level-4

frederickpek committed:
Auto Text UI Testing: add simple i/o test

frederickpek committed:
Implement Level-5

frederickpek committed:
Implement Level-6

frederickpek committed:
.gitignore: add /data/duke.txt

frederickpek committed:
Add Storage class

frederickpek committed:
Implement saving feature

frederickpek committed:
Add minimal support of LocalDate to Deadline class

frederickpek committed:
Merge branch 'branch-Level-7'

frederickpek committed:
Merge branch 'branch-Level-8'

frederickpek committed:
Use more OOP

Add UI (in place of OutputHandler), TaskList, Storage, Parser and Command classes.

Encapsulate functionality to appropriate classes.

frederickpek committed:
Organize into Packages

frederickpek committed:
Add JUnit Tests

* ToDo class
* Event class
* Deadline class
* Parser class

frederickpek committed:
Storage: change relative path for better jar usage

frederickpek committed:
Add imports

frederickpek committed:
.gitignore: change location of /data/duke.txt

frederickpek committed:
Add JavaDoc comments

frederickpek committed:
Follow the Coding Standard

Mainly change from tabs to spaces.

frederickpek committed:
Implement Find feature

frederickpek committed:
Merge branch 'branch-A-JavaDoc'

frederickpek committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/DisplayCommand.java
#	src/main/java/duke/command/ExitCommand.java
#	src/main/java/duke/command/MarkDoneCommand.java
#	src/main/java/duke/exception/DukeException.java
#	src/main/java/duke/exception/TaskException.java
#	src/main/java/duke/exception/UnknownCommandException.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskList.java
#	src/main/java/duke/task/ToDo.java
#	src/main/java/duke/util/Parser.java
#	src/main/java/duke/util/Storage.java
#	src/main/java/duke/util/UI.java

frederickpek committed:
Resolve merge conflict

frederickpek committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/DisplayCommand.java
#	src/main/java/duke/command/ExitCommand.java
#	src/main/java/duke/command/MarkDoneCommand.java
#	src/main/java/duke/util/Parser.java

frederickpek committed:
Clean up coding style

frederickpek committed:
Add JavaDoc to FindCommand

frederickpek committed:
.gitIgnore: Add data/duke.txt

frederickpek committed:
Add JavaDoc changes

frederickpek committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d' into branch-A-Gradle

frederickpek committed:
Add Gradle support and checkstyle configs

frederickpek committed:
Merge branch 'branch-A-Gradle'

frederickpek committed:
Add JavaFX tutorial 1 code

frederickpek committed:
Add JavaFX tutorial 2 code

frederickpek committed:
Add JavaFX tutorial 3: Iteration 1 - Echoing the User

frederickpek committed:
Add JavaFX tutorial 3: Iteration 2 - Adding Dialog Boxes

frederickpek committed:
Add JavaFX tutorial 3: Iteration 3 - Adding custom behaviour to DialogBox

frederickpek committed:
Add JavaFX tutorial 4: Using FXML

frederickpek committed:
Add Gui to Duke

frederickpek committed:
Merge branch 'branch-Level-10'

frederickpek commented:
# Duke
> I had a lot of fun coding Duke! - [Frederick Pek](https://github.com/frederickpek) 

Duke helps you track your tasks at hand, so they're free to do more iP/tP increments!

* ToDo Tasks
* Deadline Tasks
* Event Tasks

Why Duke?

1. Its **FREE**! :moneybag:
1. Its _EASY_! :grinning:

You can keep track of which tasks have been done!
- [x] Buy milk
- [x] Buy bread
- [x] sort array
- [ ] finish this GFMD task

Here is a ~~random~~ program that prints "Hello World!" The `main` method is shown below!
```java
public class HelloWorld {
    public static void main(String[] args) {
        System.out.println("Hello World!");
    }
}
```

yin72257 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,83 @@
+package duke;
+
+import java.io.*;
```
Perhaps consider importing only the libraries you need instead of *.

frederickpek replied:
Thank you for your comments! I have just explicitly listed down my wildcard imports to all my classes.

yin72257 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,48 @@
+package duke.command;
+
+import java.io.*;
+import java.util.*;
```
Same comment as above

yin72257 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,46 @@
+package duke.command;
+
+import java.io.*;
+import java.util.*;
```
Same here

yin72257 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,48 @@
+package duke.command;
+
+import java.io.*;
+import java.util.*;
```
Same here

yin72257 replied:
This appears in almost every file so you might consider checking them.

yin72257 reviewed (commented):
Mostly according to coding standards. Only issue is importing by * instead of individual classes.

frederickpek committed:
Improve coding style: remove wildcard imports

frederickpek committed:
Improve coding style: update naming convention to methods in UI class

frederickpek committed:
Improve GUI aesthetics

frederickpek committed:
Merge branch 'branch-Improve-GUI'

frederickpek committed:
Update README.md

Add minimal product user guide. To add more examples of command usage and expected response from Duke.

frederickpek committed:
Update README.md

leonghongfai commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+     * @return The task at index.
+     */
+     public Task get(int index) {
+          return this.tasks.get(index);
```
```suggestion
         return this.tasks.get(index);
```

frederickpek replied:
Thanks for the catch! I somehow set indentations to be 5-spaces for that particular file haha

leonghongfai commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+     * @return A new TaskList object.
+     */
+     public TaskList(List<Task> tasks) {
+          this.tasks = tasks;
```
```suggestion
         this.tasks = tasks;
```

leonghongfai commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+ * @version CS2103 AY21/22 Sem 1 iP
+ */
+public class TaskList {
+     private List<Task> tasks;
```
```suggestion
    private List<Task> tasks;
```

leonghongfai commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+     * @param index The index of the task to be marked as done.
+     */
+     public void markDone(int index) {
+          this.get(index).markDone();
```
```suggestion
         this.get(index).markDone();
```

leonghongfai reviewed (approved):
Looks good.  👍 
Some cosmetic changes requested in TaskList.java (indentations are over by one character).

frederickpek committed:
Improve coding style: adjust indentations in TaskList.java

frederickpek committed:
Change mainClassName in build.gradle to make new .jar file

frederickpek committed:
Add assertion to check for null arguments

frederickpek committed:
Refactor Parser

Parser deals with many possible cases of input. Refactor to maintain
code quality.

Lets,
* abstract task parsing details from the parse method
* avoid complicated expression in parseToDo method
* avoid complicated expression in parseDeadline method
* avoid complicated expression in parseEvent method

frederickpek committed:
Add Stream support to UI

UI handles many lines of output in lists.

Introduce streams to handle output formatting of the output lines.

frederickpek committed:
Merge pull request #1 from frederickpek/branch-A-Assertions

Add assertion to check for null arguments

frederickpek committed:
Merge branch 'master' into branch-A-CodeQuality

frederickpek committed:
Merge branch 'master' into branch-A-Streams

frederickpek committed:
Merge pull request #2 from frederickpek/branch-A-CodeQuality

Branch a code quality

frederickpek committed:
Merge branch 'master' into branch-A-Streams

frederickpek committed:
Merge pull request #3 from frederickpek/branch-A-Streams

Branch a streams

frederickpek committed:
Support sorting of tasks

The task list that Duke keeps track of can get large.

Let's add a sort command to support sorting tasks by type.

frederickpek committed:
Merge branch 'branch-C-Sort'

frederickpek committed:
Add more exception handling

'find' and 'done' commands needs to be typed correctly.

Lets,
* Ensure the those commands only have 2 tokens
* Ensure the second token is a valid integer
* Ensure the index is within range

frederickpek committed:
Fix text cut issue

frederickpek committed:
Add help message on startup

frederickpek committed:
Change user dialog colour

frederickpek committed:
Change Duke dialog colour for errors

frederickpek committed:
Add HelpCommand

frederickpek committed:
Improve task creation error-message format

frederickpek committed:
Merge branch 'branch-A-BetterGui'

frederickpek committed:
Fix indexing bug

frederickpek committed:
Add Ui.png

frederickpek committed:
Add user guide

frederickpek committed:
Merge branch 'branch-A-UserGuide'

frederickpek committed:
Set theme jekyll-theme-cayman

frederickpek committed:
Add default sample data to Duke

frederickpek committed:
Update README.md
